Pretty print of configuration errors

Users can do (unintended) configuration errors and Quarkus should provide human (not only Java developer) readable message. Currently java stacktrace is provided.

I can imagine several cases
 - decimal instead of integer
 - typo in boolean
 - typo in name for enums/constants - e.g. log level

Quarkus is using MP Config via SmallRye Config code so there could be also an enhancement on that side.

Proposed format:
```
java -Dquarkus.log.console.darken=1.5 -jar target/validation-1.0-SNAPSHOT-runner.jar
Error: Expected an integer value for 'quarkus.log.console.darken' property.
```

Current state:
```
java -Dquarkus.log.console.darken=1.5 -jar target/validation-1.0-SNAPSHOT-runner.jar
java.lang.NumberFormatException: For input string: "1.5"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at io.smallrye.config.Converters.lambda$static$18960f11$1(Converters.java:75)
	at io.smallrye.config.SmallRyeConfig.convert(SmallRyeConfig.java:133)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:76)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey_log_console_darken(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey_log_console(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey_log(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.getRunTimeConfiguration(Unknown Source)
	at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.Application.run(Application.java:213)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.Application.run(Application.java:213)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.NumberFormatException: For input string: "1.5"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at io.smallrye.config.Converters.lambda$static$18960f11$1(Converters.java:75)
	at io.smallrye.config.SmallRyeConfig.convert(SmallRyeConfig.java:133)
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:76)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey_log_console_darken(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey_log_console(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey_log(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.parseKey(Unknown Source)
	at io.quarkus.runtime.generated.RunTimeConfig.getRunTimeConfiguration(Unknown Source)
	... 4 more
```

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -35,7 +35,7 @@
         <microprofile-reactive-streams-operators.version>1.0.1</microprofile-reactive-streams-operators.version>
         <microprofile-rest-client.version>1.4.1</microprofile-rest-client.version>
         <smallrye-common.version>1.4.1</smallrye-common.version>
-        <smallrye-config.version>1.9.2</smallrye-config.version>
+        <smallrye-config.version>1.9.3</smallrye-config.version>
         <smallrye-health.version>2.2.5</smallrye-health.version>
         <smallrye-metrics.version>2.4.4</smallrye-metrics.version>
         <smallrye-open-api.version>2.0.13</smallrye-open-api.version>

--- a/core/runtime/src/main/java/io/quarkus/runtime/configuration/Substitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/configuration/Substitutions.java
@@ -13,6 +13,7 @@ import com.oracle.svm.core.threadlocal.FastThreadLocalFactory;
 import com.oracle.svm.core.threadlocal.FastThreadLocalInt;
 
 import io.smallrye.common.constraint.Assert;
+import io.smallrye.config.ConfigMappingMetadata;
 import io.smallrye.config.Expressions;
 
 /**
@@ -57,6 +58,15 @@ final class Substitutions {
     @TargetClass(className = "io.smallrye.config.ConfigMappingLoader")
     static final class Target_ConfigMappingLoader {
         @Substitute
+        static Class<?> loadClass(final Class<?> parent, final ConfigMappingMetadata configMappingMetadata) {
+            try {
+                return parent.getClassLoader().loadClass(configMappingMetadata.getClassName());
+            } catch (ClassNotFoundException e) {
+                return null;
+            }
+        }
+
+        @Substitute
         private static Class<?> defineClass(final Class<?> parent, final String className, final byte[] classBytes) {
             return null;
         }
@@ -76,6 +86,12 @@ final class Substitutions {
             } catch (NullPointerException e) {
                 return null;
             }
+        }
+
+        // This should not be called, but we substitute it anyway to make sure we remove any references to ASM classes.
+        @Substitute
+        public byte[] getClassBytes() {
+            return null;
         }
     }
 }

